File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        ans = 1

        seen = set(nums)
        for num in seen:
            if num - 1 not in seen: # start of seq found
                tmp = 1
                while num + 1 in seen:
                    tmp += 1
                    num += 1
                ans = max(ans, tmp)
        return ans

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.cpp
/*
input: string s 
want: int representing the longest substring of the input w/o dups
edge case: empty input, non engl chars , no nums, !alnums
time, space - O(n), O(n)

*/



class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        if (s.empty()) {return 0;} // does thsi work?

        set<char> seen;
        int ans = 0;
        int n = s.size();
        int l = 0;

        for (int i = 0; i < n; i++ ) {
            //if dups remove the emtry 
            while (seen.contains(s[i])) {
                seen.erase(s[l]);
                l++;
            }
            seen.insert(s[i]);
            ans = max(ans, (int)seen.size());
        }
        return ans;


    }
};